pkg/compact/compact.go

```diff
@@ -1008,12 +1016,19 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 			uniqueSources[s] = struct{}{}
 		}
 
-		if err := block.Download(ctx, cg.logger, cg.bkt, meta.ULID, bdir); err != nil {
+		tracing.DoInSpan(ctx, "compaction_block_download", func(ctx context.Context) {
+			err = block.Download(ctx, cg.logger, cg.bkt, meta.ULID, bdir)
+		}, opentracing.Tags{"block.id": meta.ULID})
+
+		if err != nil {
 			return false, ulid.ULID{}, retry(errors.Wrapf(err, "download block %s", meta.ULID))
 		}
 
 		// Ensure all input blocks are valid.
-		stats, err := block.GatherIndexHealthStats(cg.logger, filepath.Join(bdir, block.IndexFilename), meta.MinTime, meta.MaxTime)
+		var stats block.HealthStats
+		tracing.DoInSpan(ctx, "compaction_block_healthcheck", func(ctx context.Context) {
```

Comment:
Nit: Maybe `_healthstats` since the method does not check, only collects stats?
```suggestion
		tracing.DoInSpan(ctx, "compaction_block_healthstats", func(ctx context.Context) {
```

Reply:
Logging errors sounds good! I had tried it one of the earlier commits but reverted considering errors were already being logged. 
Instead of a custom function, there is an existing function, `span.LogFields(opentracing_log.Error(err))`. Maybe I could use that?

Reply:
I think the issue is that if we use our "custom" method `DoInSpan`, we do not have access to the actual span. Hence why I suggested a kind of enhanced version of the same method. Other option would be to not use `DoInSpan` and instrument the code "manually".

Reply:
Thanks.
There is a way to get the span from the current context, which I have used in my recent commit. That way, I don't have to instrument manually and can log errors with minimal changes.

Reply:
I have pushed a commit with an implementation of the former. Please let me know what you think.
thanks!